src/lib.rs

```diff
@@ -167,7 +167,11 @@ fn sign_es<P: ToKey>(data: &str, private_key_path: &P, algorithm: Algorithm) ->
         _  => panic!("Invalid hmac algorithm")
     };
 
-    sign(data, key, stp)
+    let mut signer = Signer::new(stp, &key)?;
```

Comment:
You should add a comment here explaining what you are doing and why (signatures are not compatible with other JWT implementations).
Link to the issue you found, and the Python implementation.